src/main/java/com/minecolonies/coremod/items/ItemScanTool.java

```diff
@@ -147,11 +148,21 @@ public static void saveStructure(@Nullable final World world, @Nullable final Bl
 
         final long currentMillis = System.currentTimeMillis();
         final String currentMillisString = Long.toString(currentMillis);
-        final String fileName = "/minecolonies/scans/" + LanguageHandler.format("item.scepterSteel.scanFormat", "", currentMillisString + ".nbt");
+        final String prefix = "/minecolonies/scans/";
+        final String fileName;
+        if(name == null || name.isEmpty())
+        {
+            fileName = LanguageHandler.format("item.scepterSteel.scanFormat", "", currentMillisString);
+        }
+        else
+        {
+            fileName = name;
```

Comment:
The Command framework changes all look good.
The only potential danger I see here is that there's no sanitizing of the filename to insure that it's not being used to hack the system.   Ie,   "name: ../../../../.Xauthority" or something more diabolical than that.  I would restrict the name to disallow special characters.   Maybe only allow [-.A-Za-z0-9]

Reply:
good point, will add something!

Reply:
Unfortunately, I'm not sure what a good solution for this would be.  What I proposed won't work in other languages.   But I haven't seen a good solution anywhere.

This was the best I found, but I don't like that you can't tell if it's sufficient, and it's not obvious what's being filtered.

https://stackoverflow.com/questions/1155107/is-there-a-cross-platform-java-method-to-remove-filename-special-chars/26420820#26420820

Reply:
I also looked around a bit, I think the best way is to completely ignore all special characters, numbers and letters are way enough =D